packages/web-components/src/avatar/avatar.ts

```diff
@@ -106,17 +114,65 @@ export class Avatar extends FASTElement {
   public colorId?: AvatarNamedColor | undefined;
 
   /**
-   * Sets the data-color attribute used for the visual presentation
+   * Holds the current color state
+   */
+  private currentColor: string | undefined;
+
+  constructor() {
+    super();
+
+    this.elementInternals.role = 'img';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+
+    Observable.getNotifier(this).subscribe(this);
+
+    this.generateColor();
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+
+    Observable.getNotifier(this).unsubscribe(this);
+  }
+
+  /**
+   * Handles changes to observable properties
    * @internal
+   * @param source - the source of the change
+   * @param propertyName - the property name being changed
    */
-  public generateColor(): AvatarColor | void {
-    if (!this.color) {
-      return;
+  public handleChange(source: any, propertyName: string) {
+    switch (propertyName) {
```

Comment:
Just a question about pattern since I'm not very familiar with the internals of FAST. Would it be more efficient to get notified for the changes of all attributes then filter to what we need here, or to use `colorChanged` and `colorIdChanged` handlers?

Reply:
We don't want to run for every property, so I'm scoping to take action on just these. In terms of not using the `changed` handlers which get invoked, I made that choice because they'll both run the same thing, which means duplication in the code we can condense.

Reply:
Ok, I see. Yes, you're scoping the attributes inside this method, but I asked about it because this `handleChange` method will still be called when any attribute is changed. I don't think it's costly, was just wondering if it's more efficient than using the `changed` callback.